sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/degrade/DegradeRule.java

```diff
@@ -52,47 +45,45 @@
  * </ul>
  *
  * @author jialiang.linjl
+ * @author Eric Zhao
  */
 public class DegradeRule extends AbstractRule {
 
-    @SuppressWarnings("PMD.ThreadPoolCreationRule")
-    private static ScheduledExecutorService pool = Executors.newScheduledThreadPool(
-        Runtime.getRuntime().availableProcessors(), new NamedThreadFactory("sentinel-degrade-reset-task", true));
-
     public DegradeRule() {}
 
     public DegradeRule(String resourceName) {
         setResource(resourceName);
     }
 
     /**
-     * RT threshold or exception ratio threshold count.
+     * Circuit breaking strategy (0: average RT, 1: exception ratio, 2: exception count).
      */
-    private double count;
+    private int grade = RuleConstant.DEGRADE_GRADE_RT;
 
     /**
-     * Degrade recover timeout (in seconds) when degradation occurs.
+     * Threshold count.
      */
-    private int timeWindow;
+    private double count;
 
     /**
-     * Degrade strategy (0: average RT, 1: exception ratio, 2: exception count).
+     * Recovery timeout (in seconds) when circuit breaker opens. After the timeout, the circuit breaker will
+     * transform to half-open state for trying a few requests.
      */
-    private int grade = RuleConstant.DEGRADE_GRADE_RT;
+    private int timeWindow;
 
     /**
-     * Minimum number of consecutive slow requests that can trigger RT circuit breaking.
+     * Minimum number of requests (in an active statistic time span) that can trigger circuit breaking.
      *
      * @since 1.7.0
      */
-    private int rtSlowRequestAmount = RuleConstant.DEGRADE_DEFAULT_SLOW_REQUEST_AMOUNT;
```

Comment:
Here is a missing property in new implementation. Is that acceptable in compatibility?

Reply:
It could bring breaking changes (and break the SEMVER spec) indeed. In convention, we should mark it deprecated and remove it until 2.x. But actually it's rarely used and often regarded as a "hidden" attribute (not even appeared in the dashboard)...

Reply:
Yeah you're right. But i just afraid the users who integrating manually suffering broken compatibility.😂 We can make decision here whether doing it carefully this time. Decide and record it and it would be ok.

Reply:
IMHO we could drop it (the legacy bad design) and add notes in relevant documents :)